### DISCUSSION TITLE
Cannot successfully compile

### DISCUSSION THREAD

**Discussion by Avninash** [protobuf-c/protobuf-c#576] [Q&A]
while compiling 1.4.0  in redhat-8 i am getting the error like this

Any help will be appreciated, Thanks in advance.....

[root@localhost protobuf-c-master]# make
g++ -std=c++11 -DHAVE_CONFIG_H -I.  -include ./config.h -I./protobuf-c -I. -I.   -I/usr/local/include  -g -O2 -MT protobuf-c/protoc_c_protoc_gen_c-protobuf-c.pb.o -MD -MP -MF protobuf-c/.deps/protoc_c_protoc_gen_c-protobuf-c.pb.Tpo -c -o protobuf-c/protoc_c_protoc_gen_c-protobuf-c.pb.o `test -f 'protobuf-c/protobuf-c.pb.cc' || echo './'`protobuf-c/protobuf-c.pb.cc
protobuf-c/protobuf-c.pb.cc:473:39: error: ‘CopyWithSourceCheck’ is not a member of ‘google::protobuf::Message’
     ::PROTOBUF_NAMESPACE_ID::Message::CopyWithSourceCheck,
                                       ^~~~~~~~~~~~~~~~~~~
protobuf-c/protobuf-c.pb.cc:777:39: error: ‘CopyWithSourceCheck’ is not a member of ‘google::protobuf::Message’
     ::PROTOBUF_NAMESPACE_ID::Message::CopyWithSourceCheck,
                                       ^~~~~~~~~~~~~~~~~~~
protobuf-c/protobuf-c.pb.cc:988:39: error: ‘CopyWithSourceCheck’ is not a member of ‘google::protobuf::Message’
     ::PROTOBUF_NAMESPACE_ID::Message::CopyWithSourceCheck,
                                       ^~~~~~~~~~~~~~~~~~~
make: *** [Makefile:1765: protobuf-c/protoc_c_protoc_gen_c-protobuf-c.pb.o] Error 1
[root@localhost protobuf-c-master]#

**Comment by nigelorr** [comment]
In my case, there were two problems, I'm not sure which caused this error.

1. I was linking against the debug binary in a Qt release build
2. My include path was to the top level src folder, now changed to the protobuf/src folder

Thanks for adding update on your fix @Avninash

**Comment by Avninash** [comment]
> I am seeing the same message. How did you resolve it?

protobuf-c require protobuf.
compile protobuf from sources first from "https://github.com/protocolbuffers/protobuf " and keep that location as pkg_config and compile protobuf-c
